src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -439,6 +465,21 @@ private static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAcce
             return instance ?? throw new InvalidOperationException(SR.Format(SR.Error_FailedToActivate, type));
         }
 
+        private static bool CanBindToTheseConstructorParameters(ParameterInfo[] constructorParameters, out string reason)
+        {
+            reason = string.Empty;
+            foreach (ParameterInfo p in constructorParameters)
+            {
+                if (p.IsOut || p.IsIn || p.ParameterType.IsByRef)
+                {
+                    reason = p.ParameterType.ToString();
```

Comment:
1. It's kind of odd to call this parameter `reason`.
2. In the calling code it is called `invalidPropertyName`, but this is the parameter Type name. Should this be `p.Name`?

Reply:
I think it was called `reason` previously as I had an idea to report any and all invalid parameters in the same response. I've changed this to report the name. Well spotted though, it makes much more sense to report it by name rather than the type that isn't supported.